### PR TITLE
fix: remove vendor dependency in virtualized toolchain

### PR DESCRIPTION
**Pull request by jessesuen**
Fixes the broken virtualized toolchain introduced in https://github.com/argoproj/argo-cd/pull/7920.

As part of this change, I also removed `jq` as part of the toolchain install, instead preferring to install jq like git, wget, curl, etc... using apt-get. The reason for this is because:

1. the `install-jq-linux.sh` script did not work on arm64 (the deb file extraction is not enough, there is a jqlib.so that requires a second deb to be installed which the `jq` binary will not function without).
2. jq has not been updated in 3 years and so we will very rarely need to update it, and even if we do, there likely wont be any difference in the behavior that we are concerned about (like wget/curl/git)

Signed-off-by: Jesse Suen <jesse@akuity.io>

### CODE REVIEW

**Review by alexmt — APPROVED** [review]
LGTM

File: hack/installers/install-codegen-go-tools.sh [hunk 1]
```diff
@@ -9,10 +9,12 @@ SRCROOT="$( CDPATH='' cd -- "$(dirname "$0")/../.." && pwd -P )"
 # unecessary dependencies to go.mod). We want to maintain a single source of truth for versioning
 # our binaries (either go.mod or go install <pkg>@<version>), so we use two techniques to install
 # our CLIs:
-# 1. For CLIs which are NOT vendored in go.mod, we can run `go install` with an explicit version
-# 2. For packages which we *do* vendor in go.mod, we can run `go install` from the vendor directory
+# 1. For CLIs which are NOT vendored in go.mod, we can run `go install <pkg>@<version>` with an explicit version
+# 2. For packages which we *do* vendor in go.mod, we determine version from go.mod followed by `go install` with that version
 go_mod_install() {
-    go install -mod=vendor ./vendor/$1
+    module=$(go list -f '{{.Module}}' $1 | awk '{print $1}')
```
Comment (ocraviotto) [inline hack/installers/install-codegen-go-tools.sh:15]: @jessesuen Just tried to build the test-tools-image (via `make test-tools-image`) and still seeing related issues to having this executed in docker:
```sh
+++ dirname ./installers/install-codegen-go-tools.sh
++ CDPATH=
++ cd -- ./installers/../..
++ pwd -P
+ SRCROOT=/
+ export GOBIN=//dist
+ GOBIN=//dist
+ mkdir -p //dist
+ go_mod_install github.com/gogo/protobuf/protoc-gen-gogofast
++ awk '{print $1}'
++ go list -f '{{.Module}}' github.com/gogo/protobuf/protoc-gen-gogofast
no required module provides package github.com/gogo/protobuf/protoc-gen-gogofast: go.mod file not found in current directory or any parent directory; see 'go help modules'
+ module=
```

Comment (jessesuen) [reply hack/installers/install-codegen-go-tools.sh:15]: Ah yes, I forgot to update Dockerfile to also add go.mod and go.sum. I'm testing this now.

File: test/container/Dockerfile [hunk 2]
```diff
@@ -37,6 +37,8 @@ ENV GOPATH /go
 ADD ./hack/tool-versions.sh .
 ADD ./hack/install.sh .
 ADD ./hack/installers installers
+ADD go.mod go.mod
+ADD go.sum go.sum
```
Comment (ocraviotto) [inline test/container/Dockerfile:41]: Thanks @jessesuen for pushing on. Unfortunately this change is not enough as we'll still need to set the PATH to include `/dist`, which is set by [this line in install-codegen-go-tools.sh](https://github.com/argoproj/argo-cd/blob/63e70a9f71915f2988c6117f280ad290eec758cd/hack/installers/install-codegen-go-tools.sh#L19) and holds the binary. 
For reference, I got around the initial issues with [my changes here](https://github.com/ocraviotto/argo-cd/commit/08dbd3f511bfa2b9bd241b9f29885595a3bdf8da).

Comment (jessesuen) [reply test/container/Dockerfile:41]: Thanks for the tip. Yes, I think test container should have /dist to PATH as well. I'm trying to test this but the current dockerhub outage is making this impossible. I'm also working through some arm64 specific problems with `make codegen` and I now realize that no one has ever gotten `make codegen` working on arm64.

Comment (jessesuen) [reply test/container/Dockerfile:42]: This should be resolved with PATH being set in image

**Comment by jessesuen** [comment]
OK after 46 minutes of `make codegen` on an Mac M1 Pro, I can confirm `make codegen` is working again, and better yet, working on Apple Silicon (arm64).

**Comment by ocraviotto** [comment]
Thank you @jessesuen I have used your change and all my local tests went back to working :)

**Pull request merged**